File: MiniDOMFinanziaria/Finanziaria/AreaManagers/CAreaManager.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica



Partial Public Class Finanziaria

    <Serializable> _
    Public Class CAreaManager
        Inherits CTeamManager

        <NonSerialized> _
        Private m_Pratiche As CAMPraticheCollection
        <NonSerialized> _
        Private m_TeamManagers As CAMTeamManagersCollection

        Public Sub New()
            Me.m_Pratiche = Nothing
            Me.m_TeamManagers = Nothing
        End Sub

        Public Overrides Function GetModule() As CModule
            Return Finanziaria.AreaManagers.Module
        End Function

        Public ReadOnly Property Pratiche As CAMPraticheCollection
            Get
                If (Me.m_Pratiche Is Nothing) Then
                    Me.m_Pratiche = New CAMPraticheCollection
                    Me.m_Pratiche.Initialize(Me)
                End If
                Return Me.m_Pratiche
            End Get
        End Property

        Public ReadOnly Property TeamManagers As CAMTeamManagersCollection
            Get
                If (Me.m_TeamManagers Is Nothing) Then
                    Me.m_TeamManagers = New CAMTeamManagersCollection
                    Me.m_TeamManagers.Initialize(Me)
                End If
                Return Me.m_TeamManagers
            End Get
        End Property

        Public Overrides Function GetTableName() As String
            Return "tbl_AreaManagers"
        End Function

        Protected Overrides Sub OnAfterSave(e As SystemEvent)
            MyBase.OnAfterSave(e)
            Finanziaria.AreaManagers.UpdateCached(Me)
        End Sub

    End Class

End Class

File: MiniDOMFinanziaria/Finanziaria/AreaManagers/CAreaManager.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Finanziaria;


namespace minidom
{
    public partial class Finanziaria
    {

        /// <summary>
        /// Area manager
        /// </summary>
        [Serializable]
        public class CAreaManager
            : CTeamManager
        {
            //[NonSerialized] private CAMPraticheCollection m_Pratiche;
            //[NonSerialized] private CAMTeamManagersCollection m_TeamManagers;

            /// <summary>
            /// Costruttore
            /// </summary>
            public CAreaManager()
            {
                m_Pratiche = null;
                m_TeamManagers = null;
            }

            public override bool Equals(Databases.DBObject obj)
            {
                return (obj is CAreaManager) && base.Equals(obj);
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            public override CModulesClass GetModule()
            {
                return minidom.Finanziaria.AreaManagers;
            }

            /// <summary>
            /// Restituisce la collezione delle pratiche caricate sotto il team-manager
            /// </summary>
            /// <returns></returns>
            public CAMPraticheCollection GetPratiche()
            {
                return new CAMPraticheCollection(this);
            }

            /// <summary>
            /// Restituisce la collezione dei team-manager gestiti dall'area manager
            /// </summary>
            public CAMTeamManagersCollection GetTeamManagers()
            {
                return new CAMTeamManagersCollection(this);
            }

            /// <summary>
            /// Discriminator
            /// </summary>
            /// <returns></returns>
            public override string GetTableName()
            {
                return "tbl_AreaManagers";
            }

             
        }
    }
}